api/post.go

```diff
@@ -129,7 +131,7 @@ func CreateValetPost(c *Context, post *model.Post) (*model.Post, *model.AppError
 	return rpost, nil
 }
 
-func CreatePost(c *Context, post *model.Post, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
+func CreatePost(c *Context, post *model.Post, teamId string, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
```

Comment:
We should mock the session up.  Someone changing internal code to CreatePosts should always be able to depend on the context.  If a context truly does not exist then we should build one with all the appropriate data.